File: kt/src/main/kotlin/com/github/scglwsj/parking/ticket.kt
package com.github.scglwsj.parking

import java.util.*

class Ticket(private val carPlate: PlateNumber, val parkingLotID: ParkingLotID) :
    ValueObject<TicketId>(TicketId(UUID.randomUUID().toString())) {
    var valid = true
    val isValid get() = valid
    val id get() = value
    fun invalidate() {
        valid = false
    }
}

class TicketId(id: String) : ValueObject<String>(id)

File: cs/Parking/Ticket.cs
using System;

namespace parking
{
    public class Ticket : ValueObject<TicketID>
    {
        public bool IsValid { get; private set; }
        public PlateNumber CarPlante { get; private set; }
        public ParkingLotID ParkingLotID { get; private set; }

        public Ticket(PlateNumber carPlante, ParkingLotID parkingLotID)
            : base(new TicketID(Guid.NewGuid().ToString()))
        {
            IsValid = true;
            CarPlante = carPlante;
            ParkingLotID = parkingLotID;
        }

        public TicketID ID => value;

        public void Invalidate()
        {
            IsValid = false;
        }
    }

    public class TicketID : ValueObject<string>
    {
        public TicketID(string ticketID) : base(ticketID) { }
    }
}
